src/main/java/nextstep/qna/domain/Answer.java

```diff
@@ -47,17 +51,28 @@ public Long getId() {
         return id;
     }
 
-    public Answer setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
+    public DeleteHistory deleteByUser(NsUser user) throws CannotDeleteException {
+        validateUser(user);
+        this.deleted = true;
```

Comment:
```suggestion
        this.deleted();
```

삭제는 중요하니 하나의 책임을 가지는 메소드로 분리해보는건 어떻게 생각하시나요 ? 😄